<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>ESAPI for PHP Core: ValidationException Class Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.6.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul>
      <li><a href="main.html"><span>Main&nbsp;Page</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li class="current"><a href="annotated.html"><span>Data&nbsp;Structures</span></a></li>
    </ul>
  </div>
  <div class="tabs">
    <ul>
      <li><a href="annotated.html"><span>Data&nbsp;Structures</span></a></li>
      <li><a href="hierarchy.html"><span>Class&nbsp;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Data&nbsp;Fields</span></a></li>
    </ul>
  </div>
</div>
<div class="contents">
<h1>ValidationException Class Reference</h1><!-- doxytag: class="ValidationException" --><!-- doxytag: inherits="EnterpriseSecurityException" -->
<p>Inherits <a class="el" href="class_enterprise_security_exception.html">EnterpriseSecurityException</a>.</p>

<p>Inherited by <a class="el" href="class_validation_availability_exception.html">ValidationAvailabilityException</a>, and <a class="el" href="class_validation_upload_exception.html">ValidationUploadException</a>.</p>
<div class="dynheader">
Collaboration diagram for ValidationException:</div>
<div class="dynsection">
<div class="center"><img src="class_validation_exception__coll__graph.png" border="0" usemap="#_validation_exception_coll__map" alt="Collaboration graph"/></div>
<map name="_validation_exception_coll__map" id="_validation_exception_coll__map">
<area shape="rect" id="node2" href="class_enterprise_security_exception.html" title="EnterpriseSecurityException" alt="" coords="5,6,192,34"/></map>
<center><span class="legend">[<a target="top" href="graph_legend.html">legend</a>]</span></center></div>
<table border="0" cellpadding="0" cellspacing="0">
<tr><td colspan="2"><h2>Public Member Functions</h2></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_validation_exception.html#a1b29b515c8a253e6e41c25197fd5f19f">__construct</a> ($userMessage= '', $logMessage= '', $context= '')</td></tr>
<tr><td class="memItemLeft" align="right" valign="top">&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_validation_exception.html#a51609a8c7faaf9948679e3a291ba4733">getContext</a> ()</td></tr>
<tr><td class="memItemLeft" align="right" valign="top">&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_validation_exception.html#a313783d5477383dd99a2f3785705aaf4">setContext</a> ($context)</td></tr>
</table>
<hr/><a name="_details"></a><h2>Detailed Description</h2>
<p>A <a class="el" href="class_validation_exception.html">ValidationException</a> should be thrown to indicate that the data provided by the user or from some other external source does not match the validation rules that have been specified for that data. </p>
<hr/><h2>Constructor &amp; Destructor Documentation</h2>
<a class="anchor" id="a1b29b515c8a253e6e41c25197fd5f19f"></a><!-- doxytag: member="ValidationException::__construct" ref="a1b29b515c8a253e6e41c25197fd5f19f" args="($userMessage= '', $logMessage= '', $context= '')" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">__construct </td>
          <td>(</td>
          <td class="paramtype">$&nbsp;</td>
          <td class="paramname"> <em>userMessage</em> = <code>''</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">$&nbsp;</td>
          <td class="paramname"> <em>logMessage</em> = <code>''</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">$&nbsp;</td>
          <td class="paramname"> <em>context</em> = <code>''</code></td><td>&nbsp;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td><td></td>
        </tr>
      </table>
</div>
<div class="memdoc">
<p>Instantiates a new <a class="el" href="class_validation_exception.html">ValidationException</a>.</p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table border="0" cellspacing="2" cellpadding="0">
    <tr><td valign="top"></td><td valign="top"><em>userMessage</em>&nbsp;</td><td>the message to display to users </td></tr>
    <tr><td valign="top"></td><td valign="top"><em>logMessage</em>&nbsp;</td><td>the message logged </td></tr>
    <tr><td valign="top"></td><td valign="top"><em>cause</em>&nbsp;</td><td>the cause </td></tr>
    <tr><td valign="top"></td><td valign="top"><em>context</em>&nbsp;</td><td>the source that caused this exception </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<hr/><h2>Member Function Documentation</h2>
<a class="anchor" id="a51609a8c7faaf9948679e3a291ba4733"></a><!-- doxytag: member="ValidationException::getContext" ref="a51609a8c7faaf9948679e3a291ba4733" args="()" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">getContext </td>
          <td>(</td>
          <td class="paramname"></td>
          <td>&nbsp;)&nbsp;</td>
          <td></td>
        </tr>
      </table>
</div>
<div class="memdoc">
<p>Returns the UI reference that caused this <a class="el" href="class_validation_exception.html">ValidationException</a></p>
<dl class="return"><dt><b>Returns:</b></dt><dd>context, the source that caused the exception, stored as a string </dd></dl>

</div>
</div>
<a class="anchor" id="a313783d5477383dd99a2f3785705aaf4"></a><!-- doxytag: member="ValidationException::setContext" ref="a313783d5477383dd99a2f3785705aaf4" args="($context)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">setContext </td>
          <td>(</td>
          <td class="paramtype">$&nbsp;</td>
          <td class="paramname"> <em>context</em></td>
          <td>&nbsp;)&nbsp;</td>
          <td></td>
        </tr>
      </table>
</div>
<div class="memdoc">
<p>Set's the UI reference that caused this <a class="el" href="class_validation_exception.html">ValidationException</a></p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table border="0" cellspacing="2" cellpadding="0">
    <tr><td valign="top"></td><td valign="top"><em>context</em>&nbsp;</td><td>the context to set, passed as a String </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
</div>
<hr size="1"/><address style="text-align: right;"><small>Generated on Wed Nov 18 15:13:38 2009 for ESAPI for PHP Core by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.6.1 </small></address>
</body>
</html>
